File: bending-api/src/main/java/pw/dotdash/bending/api/ability/AbilityService.java
package pw.dotdash.bending.api.ability;

import org.spongepowered.api.Sponge;
import org.spongepowered.api.scheduler.SpongeExecutorService;

public interface AbilityService {

    /**
     * Gets the singleton instance of the {@link AbilityService}.
     *
     * @return The singleton service instance
     */
    static AbilityService getInstance() {
        return Sponge.getServiceManager().provideUnchecked(AbilityService.class);
    }

    long getAbilityDelayMilli();

    SpongeExecutorService getSyncExecutor();
}

File: src/main/kotlin/arvenwood/bending/api/service/AbilityService.kt
package arvenwood.bending.api.service

import org.spongepowered.api.Sponge

interface AbilityService {

    companion object {
        @JvmStatic
        fun get(): AbilityService =
            Sponge.getServiceManager().provideUnchecked(AbilityService::class.java)
    }

    val timeStep: Long

    val abilityDelayMilli: Long
}